utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/encode/AvroSchemaOutputValidator.scala

```diff
@@ -242,32 +248,28 @@ class AvroSchemaOutputValidator(validationMode: ValidationMode) extends LazyLogg
         case _ => validationTypingResult
       }
     }
-
   }
 
-  private def validateSchemaWithBaseType[T:ClassTag](typingResult: TypingResult, schema: Schema, path: Option[String]): Validated[NonEmptyList[OutputValidatorTypeError], Unit] = {
+  private def validateSchemaWithBaseType[T:ClassTag](typingResult: TypingResult, schema: Schema, path: Option[String]): ValidatedNel[OutputValidatorError, Unit] = {
     val schemaAsTypedResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
     val clazz: Class[_] = implicitly[ClassTag[T]].runtimeClass
 
     (schemaAsTypedResult, typingResult) match {
-      case (TypedClass(schemaClass, _), TypedClass(typeClass, _)) if typeClass == schemaClass => valid
-      case (_, TypedClass(typeClass, _)) if clazz == typeClass => valid
-      case (_, TypedObjectWithValue(underlying, _)) if clazz == underlying.klass => valid
-      case _ => invalid(typingResult, schema, path)
+      case (_, typing: SingleTypingResult) if clazz == typing.objType.klass => valid
+      case _ => canBeSubclassOf(typingResult, schema, path)
     }
   }
 
-  /**
-    * TODO: Consider verification class instead of using .canBeSubclassOf from Typing - at avro we want to avoid:
-    * * Unknown.canBeSubclassOf(Any) => true
-    * * Long.canBeSubclassOf(Integer) => true
-    * Should we use strict verification at avro?
-    */
   private def canBeSubclassOf(typingResult: TypingResult, schema: Schema, path: Option[String]): ValidatedNel[OutputValidatorError, Unit] = {
-      val schemaAsTypedResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
-      condNel(typingResult.canBeSubclassOf(schemaAsTypedResult), (),
-        OutputValidatorTypeError(path.getOrElse(SimpleAvroPath), typingResult, AvroSchemaExpected(schema))
-      )
+    val schemaAsTypedResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
+    val additionalTypes = additionalTypesMapping.getOrElse(schema.getType, Set.empty)
+
+    (schemaAsTypedResult, typingResult) match {
+      case (schemaType: SingleTypingResult, typing: SingleTypingResult) if schemaType.objType.klass == typing.objType.klass => valid
+      case (_, typing: SingleTypingResult) if additionalTypes.contains(typing.objType.klass) => valid
+      case (_, Unknown) => valid //Unknown is proper for each of type
```

Comment:
I'm not sure about `Unknown`..  Right now I think, we should remove this line. WDYT @arkadius ?

Reply:
For method invocations (`#UTILS.method()`) you can pass Unknown as an argument and I assume that it can fail in the same way as in avro thanks to this line. So it looks like we are consistent in this. I would leave that with comment that we probably shouldn't accept Unknown here and in other places where some value is passed (like method invocations). But before that we must be sure that user can live without Unknowns. After introducing generic functions we will be one step closer to that...

Reply:
If you decide to leave this line, I think it should be moved from here to first line of `validateTypingResult` - it should work the same for all other cases like records, maps etc.